apps/dashboard/src/app/team/[team_slug]/(team)/~/settings/layout.tsx

```diff
@@ -1,61 +1,19 @@
-"use client";
+import { getTeams } from "@/api/team";
+import { notFound } from "next/navigation";
+import { SettingsLayout } from "./SettingsLayout";
 
-import { cn } from "@/lib/utils";
-import { usePathname } from "next/navigation";
-import { useState } from "react";
-import { TeamSettingsSidebar } from "./_components/sidebar/TeamSettingsSidebar";
-import { TeamSettingsMobileNav } from "./_components/sidebar/TeamsMobileNav";
-import { getTeamSettingsLinks } from "./_components/sidebar/getTeamSettingsLinks";
-
-// on the /~/settings page
-// - On desktop: show the general settings as usual
-// - On mobile: show the full nav instead of page content and when user clicks on the "General Settings" ( first link ) - hide the full nav and show the page content
-
-export default function Layout(props: {
+export default async function Layout(props: {
   params: {
     team_slug: string;
   };
   children: React.ReactNode;
 }) {
-  const [_showFullNavOnMobile, setShowFullNavOnMobile] = useState(true);
-  const pathname = usePathname();
-  const isSettingsOverview = (pathname || "").endsWith("/~/settings");
-  const showFullNavOnMobile = _showFullNavOnMobile && isSettingsOverview;
-  const links = getTeamSettingsLinks(props.params.team_slug);
-  const activeLink = links.find((link) => pathname === link.href);
-
-  return (
-    <div>
-      {/* Huge page title  */}
-      <div className="border-border border-b py-10">
-        <div className="container">
-          <h1 className="font-semibold text-3xl tracking-tight">
-            Team Settings
-          </h1>
-        </div>
-      </div>
+  const teams = await getTeams();
+  const team = teams.find((t) => t.slug === props.params.team_slug);
```

Comment:
should be able to get the team by the slug directly instead of fetching the list?

Reply:
yeah, I can do that. 

But I assumed that since the header component uses `getTeams` - it would be cached and shared with the page as well by next.js?

anyway, I'm changing to `getTeamBySlug`

Reply:
hm yeah you're right if it's the same request but actually maybe the header should lazy load that and only also fetch the current team (since it only needs to display other teams on click)? - anyways more specific = better to begin with